app/assets/javascripts/controllers/alerts/alerts_list_controller.js

```diff
@@ -0,0 +1,132 @@
+/* global miqHttpInject */
+
+angular.module('alertsCenter').controller('alertsListController',
+  ['$window', 'alertsCenterService', '$interval', '$timeout',
+  function($window, alertsCenterService, $interval, $timeout) {
+    var vm = this;
+
+    vm.alerts = [];
+    vm.alertsList = [];
+
+    function processData(response) {
+      var updatedAlerts = alertsCenterService.convertToAlertsList(response);
+
+      // update display data for the alerts from the current alert settings
+      angular.forEach(updatedAlerts, function(nextUpdate) {
+        matchingAlert = _.find(vm.alerts, function(existingAlert) {
+          return nextUpdate.id === existingAlert.id;
+        });
+
+        if (angular.isDefined(matchingAlert)) {
+          nextUpdate.isExpanded = matchingAlert.isExpanded;
+        }
+      });
+
+      vm.alerts = updatedAlerts;
+      vm.loadingDone = true;
+      vm.filterChange();
+
+      $timeout();
```

Comment:
Aah, but that looks like the wrong place to solve this, since the digest cycle will always run if this was triggered by angular code.

Seems like all the places that call `processData` are a `then` after `API.get`, right? So, this should not be needed... Unless there's some place where this is called from non-angular context? (But in that case we need to fix that place, not throw `$timeout` around randomly..)